File: packages/public-sdk-py/blackroad/captions.py
"""Caption resource helpers."""
from __future__ import annotations

from typing import Any, Awaitable, Callable, Dict

RequestFn = Callable[[str, Dict[str, Any]], Awaitable[Any]]


class CaptionsResource:
    """Interface for caption endpoints."""

    def __init__(self, requester: RequestFn) -> None:
        self._request = requester

    async def create(self, payload: Dict[str, Any]) -> Any:
        return await self._request("/v1/captions", {"method": "POST", "json": payload})

    async def get(self, job_id: str) -> Any:
        return await self._request(f"/v1/captions/{job_id}")

    async def list_artifacts(self, job_id: str) -> Any:
        return await self._request(f"/v1/captions/{job_id}/artifacts")

File: packages/public-sdk-js/src/resources/captions.ts
import type { RequestInit } from 'node-fetch';
import type { JsonFetcher } from './client.js';

export interface CaptionJob {
  jobId: string;
  status: 'queued' | 'processing' | 'completed' | 'failed';
  submittedAt?: string;
  artifacts?: CaptionArtifact[];
}

export interface CaptionArtifact {
  type: 'vtt' | 'srt' | 'txt';
  url: string;
  checksum?: string;
}

export interface CreateCaptionJobInput {
  assetUrl: string;
  sourceLanguage: string;
  targetLanguages: string[];
  backend?: 'native' | 'premium' | 'experimental';
}

export class CaptionsResource {
  constructor(private readonly fetcher: JsonFetcher) {}

  create(input: CreateCaptionJobInput): Promise<{ jobId: string; status: string; estimatedCompletionSeconds: number }> {
    const init: RequestInit = {
      method: 'POST',
      body: JSON.stringify(input),
    };
    return this.fetcher('/v1/captions', init) as Promise<{
      jobId: string;
      status: string;
      estimatedCompletionSeconds: number;
    }>;
  }

  get(jobId: string): Promise<CaptionJob> {
    return this.fetcher(`/v1/captions/${jobId}`) as Promise<CaptionJob>;
  }

  listArtifacts(jobId: string): Promise<{ jobId: string; artifacts: CaptionArtifact[] }> {
    return this.fetcher(`/v1/captions/${jobId}/artifacts`) as Promise<{ jobId: string; artifacts: CaptionArtifact[] }>;
  }
}
